src/eko/output/struct.py

```diff
@@ -0,0 +1,502 @@
+            stream = tar.extractfile(name)
+
+            if stream is None:
+                raise ValueError
+
+            op = Operator.load(stream, compressed=compressed)
+
+        self._operators[q2] = op
+        return op
+
+    def __setitem__(self, q2: float, op: Optional[Operator], compress: bool = True):
+        if op is not None:
+            stream, without_err = op.save(compress)
+
+            suffix = "npy" if without_err else "npz"
+            if compress:
+                suffix += ".lz4"
+
+            info = tarfile.TarInfo(name=f"operators/{q2:8.2f}.{suffix}")
+            info.size = len(stream.getbuffer())
+            info.mtime = time.time()
+            info.mode = 436
+            #  info.uname = os.getlogin()
+            #  info.gname = os.getlogin()
+
+            with tarfile.open(self.path, "a") as tar:
+                tar.addfile(info, fileobj=stream)
+
+        self._operators[q2] = op
+
+    def __delitem__(self, q2: float):
+        """Drop operator from memory.
+
+        This method only drops the operator from memory, and it's not expected
+        to do anything else.
+
+        Autosave is done on set, and explicit saves are performed by the
+        computation functions.
+
+        If a further explicit save is required, repeat explicit assignment::
+
+            eko[q2] = eko[q2]
+
+        This is only useful if the operator has been mutated in place, that in
+        general should be avoided, since the operator should only be the result
+        of a full computation or a library manipulation.
+
+
+        Parameters
+        ----------
+        q2 : float
+            the value of :math:`Q^2` for which the corresponding operator
+            should be dropped
+
+        """
+        self._operators[q2] = None
+
+    @contextlib.contextmanager
+    def operator(self, q2: float):
+        try:
+            yield self[q2]
+        finally:
+            del self[q2]
+
+    @property
+    def Q2grid(self):
+        return np.array(list(self._operators))
+
+    def __iter__(self):
+        """Iterate over keys (i.e. Q2 values)
+
+        Yields
+        ------
+        float
+            q2 values
+
+        """
+        for q2 in self._operators:
+            yield q2
+
+    def items(self):
+        """Iterate operators, with minimal load.
+
+        Pay attention, this iterator:
+
+        - is not a read-only operation from the point of view of the in-memory
+          object (since the final result after iteration is no operator loaded)
+        - but it is a read-only operation from the point of view of the
+          permanent object on-disk
+
+        Yields
+        ------
+        tuple
+            couples of ``(q2, operator)``, loaded immediately before, unloaded
+            immediately after
+
+        """
+        for q2 in self.Q2grid:
+            yield q2, self[q2]
+            del self[q2]
+
+    def __contains__(self, q2: float) -> bool:
```

Comment:
this is complicated because `__contains__` and `items()` do not correlate (on in memory the other on disk)

Reply:
postponed to #139